File: Servidor/EventManagement/src/main/java/com/lausebscode/modelo/FeriaGastro.java
package com.lausebscode.modelo;

import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.PositiveOrZero;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;
import java.time.LocalDateTime;
import java.util.List;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@SuperBuilder
public class FeriaGastro {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @NotBlank(message = "El nombre no puede estar vacío")
    private String nombre;

    @NotNull
    @PositiveOrZero(message = "El precio no puede ser negativo")
    private double precio;

    @NotNull(message = "La fecha de realización no puede estar vacía")
    private LocalDateTime fechaRealizacion;

    @NotBlank(message = "El tipo no puede estar vacío")
    private String tipo;

    @OneToMany(mappedBy = "feriaGastro", cascade = {CascadeType.PERSIST, CascadeType.MERGE}, orphanRemoval = false)
    @JsonManagedReference
    private List<Organizador> organizadores;
}

File: Cliente/dotnet-eventmng-p3/dotnet-eventmng-p3/Modelo/FeriaGastro.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace dotnet_eventmng_p3.Modelo
{
    public class FeriaGastro
    {
        public int id { get; set; }
        public string nombre { get; set; }
        public double precio { get; set; }
        public DateTime fechaRealizacion { get; set; }
        public string tipo { get; set; }
        public List<int> organizadorIds { get; set; }

        public FeriaGastro(int id, string nombre, double precio, DateTime fechaRealizacion, string tipo, List<int> organizadorIds)
        {
            this.id = id;
            this.nombre = nombre;
            this.precio = precio;
            this.fechaRealizacion = fechaRealizacion;
            this.tipo = tipo;
            this.organizadorIds = organizadorIds;
        }
    }
}
